src/Rules/Test.hs

```diff
@@ -160,4 +160,15 @@ needfile stage pkg
 -- we are going to use, I suppose?
     | isLibrary pkg = pkgConfFile (Context stage pkg profilingDynamic)
     | otherwise = programPath =<< programContext stage pkg
-
+ 
+getCompiler :: Action FilePath
+getCompiler = do
+    args            <- userSetting defaultTestArgs
+    compilerPath $ testCompiler args
+
+-- | Set Test Compiler
+compilerPath :: String -> Action FilePath
```

Comment:
I'm a bit unhappy about using strings here. Could the first argument be turned into a `Stage`, ideally during parsing of test arguments?

Reply:
Well, I suppose this could be turned into `compilerPath :: Either Stage FilePath -> Action FilePath`  ?

Reply:
@alpmestan Indeed, I missed the last case.

Reply:
@chitrak7 The benefit is in keeping parsing, string manipulation and associated errors in one place. The rest of the codebase should preferably be strongly typed.